app/controllers/api/braintree_controller.rb

```diff
@@ -1,31 +1,57 @@
 class Api::BraintreeController < ApplicationController
+  skip_before_action :verify_authenticity_token
+
 
   def token
     render json: { token: ::Braintree::ClientToken.generate }
   end
 
   def transaction
-    sale = braintree.make_transaction(options)
+    result = braintree::Transaction.make_transaction(transaction_options)
+    if result.success?
+      render json: { success: true, transaction_id: result.transaction.id }
+    else
+      render json: { success: false, errors: result.errors }
```

Comment:
Yeah the difference is that from the javascript side, there are two different callbacks, one for the success case, one for the error case. If we return 200 for errors, then the wrong callback gets called.